imapserver/imapmemserver/mailbox.go

```diff
@@ -32,9 +33,28 @@ func NewMailbox(name string, uidValidity uint32) *Mailbox {
 		uidValidity: uidValidity,
 		name:        name,
 		uidNext:     1,
+		rights:      map[imap.RightsIdentifier]imap.RightSet{},
 	}
 }
 
+func (mbox *Mailbox) Rights() map[imap.RightsIdentifier]imap.RightSet {
+	mbox.mutex.Lock()
+	defer mbox.mutex.Unlock()
+
+	rightsCopy := make(map[imap.RightsIdentifier]imap.RightSet, len(mbox.rights))
+	for k, v := range mbox.rights {
+		rightsCopy[k] = v
+	}
+
+	return rightsCopy
+}
+
+func (mbox *Mailbox) SetRights(identifier imap.RightsIdentifier, rights imap.RightSet) {
+	mbox.mutex.Lock()
+	mbox.rights[identifier] = rights
+	mbox.mutex.Unlock()
+}
```

Comment:
Can we not export these methods in our public API for now? I'm not super confident this is what we'll want to expose long-term.